index.js

```diff
@@ -0,0 +1,76 @@
+const fs = require("fs");
+const React = require("react");
+const ReactDOMServer = require("react-dom/server");
+
+const App = require("./app");
+
+const getDirectories = (path, ignore = []) =>
+  fs
+    .readdirSync(path)
+    .filter(a => fs.statSync(`${path}${a}`).isDirectory())
+    .filter(a => !ignore.includes(a));
+
+const getFiles = (path, extension, ignore = []) =>
+  fs
+    .readdirSync(path)
+    .filter(a => fs.statSync(`${path}${a}`).isFile())
+    .filter(a => a.endsWith(extension))
+    .filter(a => !ignore.includes(a))
+    .map(a => {
+      const content = fs.readFileSync(`${path}${a}`).toString();
+      return {
+        name: a,
+        content,
+        headline: content.match(/# \[(.*?)\]/)[1],
+        dateAdded: fs.statSync(`${path}${a}`).birthtime
+      };
+    });
+
+const directories = getDirectories("./", [".git", "node_modules"]);
+
+const markdownFiles = directories.reduce(
+  (acc, s) =>
+    Object.assign({}, acc, {
+      [s]: getFiles(`./${s}/`, ".md")
+    }),
+  {}
+);
+
+const latestFiles = directories
+  .reduce((acc, s) => [...acc, ...getFiles(`./${s}/`, ".md")], [])
+  .sort((a, b) => {
+    const diff = a.dateAdded - b.dateAdded;
+    if (diff === 0) return diff;
+    return diff > 0 ? -1 : 1;
+  })
+  .splice(0, 3);
```

Comment:
It's fine, honestly, but technically `splice` is a mutation

Reply:
Damn, I think I meant `slice`